docs/design/data_plane_manager.adoc

```diff
@@ -0,0 +1,1189 @@
+. When heart beat loss decteted (via Netdata) , data plane manager would send out rescue message to MessagingQ to let adjacent ovs do remote rescue
+
+==== Problems solution:
+
+. To solve the misorder data, put them in same queue
+. Use the tag to filter msg.
+. RocketMQ could support way more topic number with stable performance, we need to fine tune the parameter for dirty page and memory usage
+. Use bond to get more bandwidth for storm
+image:images/bond.png[image,width=400,height=240]
+
+[Figure 4] IP Bond
+
+. Since Kafka would drop lots of performance when topic number is higher, RocketMQ is stable
+image:images/dpm-mq3.png[image,width=550,height=270]
+
+[Figure 5] Performance drop when topics nubmer is greater
+
+image:images/dpm-mq1.png[image,width=400,height=240]
+
+[Figure 6] Performance drop when topics nubmer is greater
+
+. When message body is greater than 4k , the performance would drop greatly
+image:images/dpm-mq2.png[image,width=400,height=240]
+
+[Figure 7] Message size and the performance drop
+
+. Here is the Neutron way to start ovs and do processing
+image:images/neutron-dpm.png[image,width=660,height=384]
+
+[Figure 8] Neutron way to start ovs and do processing
+
+. GRPC is not fit for transfering big file since it basically focus more on the assumption of small numbers in the data
+
+== Database Schema
+
+[cols=",,,",options="header",]
+|===
+|Vpc_states |Subnet_ states |Port_ states |security_group_ states
+| | | |
+|===
+
+State could contains more same type data as array
+
+* DataPlaneManager would deploy the goal state configuration in batch to the ACA grouply in parallel then wait for the response and do next accordingly.
+
+== REST APIs
+
+=== API Snapshot
+
+[width="100%",cols="22%,12%,50%,17%"]
+|===
+|*API Name* |*Method* |*Request*|*Response*
+
+|Get GoalState
+|GET
+|/goalstate/{goalstateid}
+|GoalState Info
+<<Get_GoalState,[sample]>>
+
+|Create Port by GoalState
+|POST
+|/goalstate/{gsinfo}
+|Goal State list
+<<Create_Port_GoalState,[sample]>>
+
+|Delete Port by GoalState
+|DELETE
+|/goalstate/{gsinfo}
+|GOAL State
+<<Delete_Port_GoalState,[sample]>>
+
+|Update GoalState
+|PUT
+|/goalstate/{goalstates}
+|GOAL State
+<<Update_GoalState,[sample]>>
+
+|Bulk Create Port by GoalState
+|POST
+|/goalstate/{gsinfo}
+|Goal State list
+<<BULK_Create_Port_GoalState,[sample]>>
+
+|Bulk Delete Port by GoalState
+|DELETE
+|/goalstate/{gsinfo}
+|GOAL State
+<<BULK_Delete_Port_GoalState,[sample]>>
+
+|Create Vpc by GoalState
+|POST
+|/goalstate/{gsinfo}
+|Goal State list
+<<Create_Vpc_GoalState,[sample]>>
+
+|Delete Vpc by GoalState
+|DELETE
+|/goalstate/{gsinfo}
+|GOAL State
+<<Delete_Vpc_GoalState,[sample]>>
+
+|Bulk Create Vpc by GoalState
+|POST
+|/goalstate/{gsinfo}
+|Goal State list
+<<BULK_Create_Vpc_GoalState,[sample]>>
+
+|Bulk Delete Vpc by GoalState
+|DELETE
+|/goalstate/{gsinfo}
+|GOAL State
+<<BULK_Delete_Vpc_GoalState,[sample]>>
+
+|===
+
+=== API Specification
```

Comment:
Use this link to review the design. Currently the format remains unorganized. https://github.com/haboy52581/alcor/blob/master/docs/design/data_plane_manager.adoc#communication-with-agent

Other microservice has established good API docs, for example, private IP and mac managers. Please take a look.
https://github.com/futurewei-cloud/alcor/blob/master/docs/modules/ROOT/pages/mgmt_services/private_ip_manager.adoc